### DISCUSSION TITLE
libvips c++: write_to_buffer takes very long time

### DISCUSSION THREAD

**Discussion by yanboma** [libvips/libvips#3390] [Q&A]
a tiff file , more than 2G。 after getting the thumbnail through VImage "thumbnail_image", call write_to_buffer(".jpg", &buf, &buf_size); the function takes almost 2 hours.

v_image = VImage::tiffload(tmpSrcFileName.c_str());
thumbnail_image = v_image.thumbnail_image(500);
void *buf = nullptr; size_t buf_size;
**thumbnail_image.write_to_buffer(".jpg", &buf, &buf_size);**

**Comment by jcupitt** [comment]
Hi @yanboma,

Don't use `thumbnail_image` -- it's only there for emergencies. It can't use any of the file format tricks to speed up thumbnailing.

Try:

```C++
thumbnail_image = VImage::thumbnail(tmpSrcFileName.c_str(), 500);
void *buf = nullptr; size_t buf_size;
thumbnail_image.write_to_buffer(".jpg", &buf, &buf_size);
```

Some large TIFFs are just very slow to decode, you might have hit one of those. What do you see for `tiffinfo myfile.tif`?
